src/main/java/org/prebid/server/floors/BasicPriceFloorEnforcer.java

```diff
@@ -0,0 +1,213 @@
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.ThreadLocalRandom;
+
+public class BasicPriceFloorEnforcer implements PriceFloorEnforcer {
+
+    private final CurrencyConversionService currencyConversionService;
+
+    public BasicPriceFloorEnforcer(CurrencyConversionService currencyConversionService) {
+        this.currencyConversionService = Objects.requireNonNull(currencyConversionService);
+    }
+
+    @Override
+    public AuctionParticipation enforce(BidRequest bidRequest,
+                                        AuctionParticipation auctionParticipation,
+                                        Account account) {
+
+        final AccountPriceFloorsConfig accountPriceFloorsConfig = ObjectUtil.getIfNotNull(account.getAuction(),
+                AccountAuctionConfig::getPriceFloors);
+
+        return shouldApplyEnforcement(auctionParticipation, accountPriceFloorsConfig)
+                ? applyEnforcement(bidRequest, auctionParticipation, accountPriceFloorsConfig)
+                : auctionParticipation;
+    }
+
+    private static boolean shouldApplyEnforcement(AuctionParticipation auctionParticipation,
+                                                  AccountPriceFloorsConfig accountPriceFloorsConfig) {
+
+        if (accountPriceFloorsConfig == null || BooleanUtils.isNotTrue(accountPriceFloorsConfig.getEnabled())) {
+            return false;
+        }
+
+        final BidderRequest bidderRequest = auctionParticipation.getBidderRequest();
+        final BidRequest bidderBidRequest = ObjectUtil.getIfNotNull(bidderRequest, BidderRequest::getBidRequest);
+        final ExtRequestPrebidFloors floors = getFloors(bidderBidRequest);
+        final ExtRequestPrebidFloorsEnforcement enforcement = ObjectUtil.getIfNotNull(floors,
+                ExtRequestPrebidFloors::getEnforcement);
+
+        return isEnforcedByRequest(enforcement) && isSatisfiedByEnforceRate(enforcement);
+    }
+
+    private static ExtRequestPrebidFloors getFloors(BidRequest bidRequest) {
+        final ExtRequest ext = ObjectUtil.getIfNotNull(bidRequest, BidRequest::getExt);
+        final ExtRequestPrebid prebid = ObjectUtil.getIfNotNull(ext, ExtRequest::getPrebid);
+
+        return ObjectUtil.getIfNotNull(prebid, ExtRequestPrebid::getFloors);
+    }
+
+    private static boolean isEnforcedByRequest(ExtRequestPrebidFloorsEnforcement enforcement) {
+        final Boolean enforcePbs = ObjectUtil.getIfNotNull(enforcement,
+                ExtRequestPrebidFloorsEnforcement::getEnforcePbs);
+
+        return BooleanUtils.isNotFalse(enforcePbs);
+    }
+
+    private static boolean isSatisfiedByEnforceRate(ExtRequestPrebidFloorsEnforcement enforcement) {
+        final Integer enforceRate = ObjectUtil.getIfNotNull(enforcement,
+                ExtRequestPrebidFloorsEnforcement::getEnforceRate);
+
+        return enforceRate == null || (enforceRate >= 0 && enforceRate <= 100
```

Comment:
False result in this condition(enforceRate >= 0 && enforceRate <= 100) is config mistake. Should we log it?

Reply:
In this case `enforceRate` is part of bid request, not account config.